modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -131,7 +131,11 @@ def __init__(
 
     def _validate_axes_lengths(self):
         """Validate that labels are split correctly if split is known."""
-        if self._row_lengths_cache is not None and len(self.index) > 0:
+        if (
+            self._row_lengths_cache is not None
```

Comment:
In the test that I wrote, `_row_lengths_cache` cache is not overwritten, so the index materializes, I think here we should also check whether the index is materialized.

Same for columns.